### PR TITLE
Add 242 build target

### PR DESCRIPTION
**Pull request by rli**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .run/generateConfigs.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ class IdeVariant:
 </component>'''
 
 if __name__ == '__main__':
-    mvs = ["2023.2", "2023.3", "2024.1"]
+    mvs = ["2023.2", "2023.3", "2024.1", "2024.2"]
```
Comment (manodnyab) [inline .run/generateConfigs.py:44]: Can we also add this to the prelease GH workflow?

File: buildspec/linuxTests.yml [hunk 2]
```diff
@@ -13,11 +13,15 @@ env:
 phases:
   install:
     commands:
+      - export JAVA_HOME=/usr/lib/jvm/java-21-amazon-corretto.x86_64
```
Comment (manodnyab) [inline buildspec/linuxTests.yml:16]: Should these 2 statements be under an if-else block?

File: buildspec/windowsTests.yml [hunk 3]
```diff
@@ -8,6 +8,15 @@ env:
 phases:
   install:
     commands:
+      # need for 232 tests to pass
```
Comment (manodnyab) [inline buildspec/windowsTests.yml:11]: can we add a TODO before it, to remove these once we drop 232?

File: buildspec/windowsTests.yml [hunk 4]
```diff
@@ -31,12 +40,12 @@ phases:
         }
 
         # Rider is very expensive (spikes our CI jobs to 50% CPU, so let it do the prep work in parallel, but run tests later
-        ./gradlew -PideProfileName="$Env:ALTERNATIVE_IDE_PROFILE_NAME" check :plugin-toolkit:jetbrains-rider:compileTestKotlin --info --console plain --continue --scan --stacktrace
+        ./gradlew -PideProfileName="$Env:ALTERNATIVE_IDE_PROFILE_NAME" check :plugin-toolkit:jetbrains-rider:compileTestKotlin -x :plugin-toolkit:jetbrains-rider:check --info --console plain --continue --scan --stacktrace
```
Comment (manodnyab) [inline buildspec/windowsTests.yml:43]: Why are we running rider tests twice?

File: plugins/toolkit/jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cloudwatch/logs/LogGroupSearchActorTest.kt [hunk 5]
```diff
@@ -100,11 +101,9 @@ class LogGroupSearchActorTest : BaseCoroutineTest() {
     }
 
     @Test
-    fun loadInitialRangeThrows() {
-        runBlocking {
-            actor.channel.send(CloudWatchLogsActor.Message.LoadInitialRange(LogStreamEntry("@@@", 0), Duration.ofMillis(0)))
-            waitForTrue { actor.channel.isClosedForSend }
-        }
+    fun loadInitialRangeThrows() = runTest {
```
Comment (manodnyab) [inline plugins/toolkit/jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cloudwatch/logs/LogGroupSearchActorTest.kt:104]: Is the runTest call required here?

File: .run/generateConfigs.py (hunk 1 above)
Comment (rli) [reply .run/generateConfigs.py:44]: let's do separately

File: buildspec/linuxTests.yml (hunk 2 above)
Comment (rli) [reply buildspec/linuxTests.yml:16]: no

File: buildspec/windowsTests.yml (hunk 4 above)
Comment (rli) [reply buildspec/windowsTests.yml:43]: L43 prevents rider tests from running

File: buildspec/linuxTests.yml (hunk 2 above)
Comment (rli) [reply buildspec/linuxTests.yml:16]: this change has propagated to prod infra

File: plugins/toolkit/jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/JavaTestUtils.kt [hunk 6]
```diff
@@ -47,7 +49,12 @@ import java.nio.file.Paths
 import kotlin.io.path.isDirectory
 
 fun HeavyJavaCodeInsightTestFixtureRule.setUpJdk(jdkName: String = "Real JDK"): String {
-    val jdkHome = IdeaTestUtil.requireRealJdkHome()
+    // attempt to find a JDK that works for gradle
+    val jdkHome = JavaSdk.getInstance().suggestHomePaths().firstOrNull {
+        val version = SdkVersionUtil.getJdkVersionInfo(it)?.version ?: return@firstOrNull false
+        version < GradleJvmSupportMatrix.getAllSupportedJavaVersionsByIdea().max()
+    } ?: IdeaTestUtil.requireRealJdkHome()
+    println("Using $jdkHome as JDK home")
```
Comment (manodnyab) [inline plugins/toolkit/jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/JavaTestUtils.kt:57]: nit: can we remove this println?

File: plugins/toolkit/jetbrains-gateway/tst/software/aws/toolkits/jetbrains/gateway/connection/ScpCommandLineTest.kt [hunk 7]
```diff
@@ -114,9 +114,11 @@ class ScpCommandLineTest {
         val paths = mutableListOf<Path?>()
         attachScpListener(paths)
 
-        ExecUtil.execAndGetOutput(
-            this.knownHostsLocation(tempFolder.newFile().toPath())
-                .constructCommandLine()
+        println(
```
Comment (manodnyab) [inline plugins/toolkit/jetbrains-gateway/tst/software/aws/toolkits/jetbrains/gateway/connection/ScpCommandLineTest.kt:117]: is this println required?

**Review by manodnyab — APPROVED** [review]

File: plugins/toolkit/jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/JavaTestUtils.kt (hunk 6 above)
Comment (rli) [reply plugins/toolkit/jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/JavaTestUtils.kt:57]: useful for debugging test failure

File: plugins/toolkit/jetbrains-gateway/tst/software/aws/toolkits/jetbrains/gateway/connection/ScpCommandLineTest.kt (hunk 7 above)
Comment (rli) [reply plugins/toolkit/jetbrains-gateway/tst/software/aws/toolkits/jetbrains/gateway/connection/ScpCommandLineTest.kt:117]: useful for debugging test failure

**Pull request merged**